sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,401 @@
+            }
+
+            "-c" | "--comment" => final_args.comment = true,
+
+            x => {
+                return error::Usage {
+                    message: format!("Unknown option '{}'", x),
+                }
+                .fail()
+            }
+        }
+    }
+
+    Ok(final_args)
+}
+
+/// Query the API for the certificate bundles, returns a tuple with trusted
+/// and distrusted PEM certificates
+async fn get_certificate_bundles<P>(
+    socket_path: P,
+) -> Result<(Vec<x509_parser::pem::Pem>, Vec<x509_parser::pem::Pem>)>
+where
+    P: AsRef<Path>,
+{
+    debug!("Querying the API for settings");
+
+    let method = "GET";
+    let uri = API_SETTINGS_URI;
+    let (_code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
+        .await
+        .context(error::APIRequest { method, uri })?;
+
+    // Build a Settings struct from the response string
+    debug!("Deserializing response");
+    let settings: model::Settings =
+        serde_json::from_str(&response_body).context(error::ResponseJson { method, uri })?;
+
+    split_bundles(settings.pki.unwrap_or_default())
+}
+
+/// Returns a tuple with two lists, for trusted and distrusted certificates
+fn split_bundles(
+    certificates_bundle: HashMap<Identifier, model::PEMCertificate>,
+) -> Result<(Vec<x509_parser::pem::Pem>, Vec<x509_parser::pem::Pem>)> {
+    let mut trusted: Vec<x509_parser::pem::Pem> = Vec::new();
+    let mut distrusted: Vec<x509_parser::pem::Pem> = Vec::new();
+
+    for (name, bundle) in certificates_bundle.iter() {
+        let data = bundle.data.clone().unwrap_or_default();
+
+        // Empty data means the certificate bundle was disabled in the API
+        if data.trim() == "" {
+            debug!("Found empty bundle: {}", name);
+            continue;
+        }
+
+        let name = name.as_ref();
+        let decoded = base64::decode(data.as_bytes()).context(error::Base64Decode { name })?;
+        // Each record in the API could include one or more certificates
+        let mut pems = pems_from_iter(x509_parser::pem::Pem::iter_from_buffer(&decoded))?;
+
+        // `trusted` defaults to false if not set in the API record
+        if bundle.trusted.unwrap_or(false) {
+            trusted.append(&mut pems);
+        } else {
+            distrusted.append(&mut pems);
+        }
+    }
+
+    Ok((trusted, distrusted))
+}
+
+/// Updates the trusted certificates store, removing the distrusted certificates
+/// from the final bundle
+fn update_trusted_store<P>(
+    (mut trusted, distrusted): (Vec<x509_parser::pem::Pem>, Vec<x509_parser::pem::Pem>),
+    trusted_store: P,
+    source_bundle: P,
+    add_comment: bool,
+) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let source_bundle = source_bundle.as_ref();
+    let trusted_store = trusted_store.as_ref();
+
+    // The default bundle includes the certificates shipped with the OS
+    let default_bundle = fs::File::open(source_bundle).context(error::ReadFile {
```

Comment:
Let's have a more specific read error, to match the more specific write error.
```suggestion
    let default_bundle = fs::File::open(source_bundle).context(error::ReadSourceBundle {
```